test/lib/util.test.js

```diff
@@ -47,24 +47,55 @@ describe('download_image', function() {
         expect(asset._image_data).is.not.null;
     });
 
-    it('can handle data urls correctly', function() {
-        let imageUrl = fs.readFileSync(__dirname + '/test_files/base64_encoded_image.txt');
-        let image = fs.readFileSync(__dirname + '/test_files/base64_encoded_image.png');
+    describe('data urls', function() {
+        it('can handle png urls correctly', function() {
+            const imageUrlFile = fs.readFileSync(__dirname + '/test_files/base64_encoded_image.png.txt');
+            const image = fs.readFileSync(__dirname + '/test_files/base64_encoded_image.png');
+            const imageSha256Hash = 'e7f555cc474b1d12a52cb9b9fa012a3502e8a2d972e19d49a898629937ac13ca'
 
-        // We want this as a string
-        imageUrl = imageUrl.toString();
+            // We want this as a string
+            let imageUrl = imageUrlFile.toString();
 
-        if (imageUrl == undefined || imageUrl.length <= 0 ||
-            image == undefined || image.length <= 0) {
-          throw new Error("Invalid data loaded from test image");
-        }
+            if (imageUrl == undefined || imageUrl.length <= 0 ||
+                image == undefined || image.length <= 0) {
+              throw new Error("Invalid data loaded from test image");
+            }
 
-        let image_tag = "<img src=" + imageUrl + ">test</img>";
+            const imageTag = "<img src=" + imageUrl + ">test</img>";
 
-        const asset = util.download_img(image_tag, '');
-        expect(asset.asset_id).is.not.null;
-        expect(asset.asset_id).is.not.undefined;
-        expect(asset._image_data).to.deep.equal(image);
-        expect(asset._content_type).to.equal('image/png');
+            const asset = util.download_img(imageTag, 'a://b.com/c d.html');
+            expect(asset.asset_id).is.not.null;
+            expect(asset.asset_id).is.not.undefined;
+            expect(asset._image_data).to.deep.equal(image);
+            expect(asset._content_type).to.equal('image/png');
+            expect(asset._canonical_uri).to.equal("data:image/png;" +
```

Comment:
generally I'm against testing private variables like this -- what about turning `_to_metadata()` public and testing these fields in its output?